include/fc/io/cfile.hpp

```diff
@@ -145,6 +145,10 @@ class cfile_datastream {
 
    bool get( char& c ) { return read(&c, 1); }
 
+   size_t tellp() const {
+      return static_cast<size_t>(cf.tellp());
```

Comment:
Can you add a check for `-1` to `cfile::tellp` and throw an exception with `errno` info. That way this will not return `static_cast<size_t>(-1)`.